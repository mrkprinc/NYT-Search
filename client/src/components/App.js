import React from 'react';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import {SearchPage} from './index';
import './App.css';

class App extends React.Component {
  render() {
    return (
      <Router>
        <main>
          <h1>New York Times Search</h1>
          <h3>find, save &amp; annotate articles</h3>
          <Switch>
            <Route path='/search' component={SearchPage} />
          </Switch>
        </main>
      </Router>
    );
  }
}

export default App;
