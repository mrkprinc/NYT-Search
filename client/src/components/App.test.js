import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

const wrapper = shallow(<App />);

it('should render without exploding', () => {
  expect(wrapper).toMatchSnapshot()
})