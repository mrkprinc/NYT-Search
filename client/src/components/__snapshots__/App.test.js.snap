// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <header>
          <h1>
            New York Times Search
          </h1>
          <h3>
            find, save & annotate articles
          </h3>
          <nav>
            <Link
              replace={false}
              to="/"
            >
              Article Search
            </Link>
            <Link
              replace={false}
              to="/saved"
            >
              Saved Articles
            </Link>
          </nav>
        </header>
        <Switch>
          <Route
            component={[Function]}
            path="/saved"
          />
          <Route
            component={[Function]}
            path="/"
          />
        </Switch>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1>
              New York Times Search
            </h1>
            <h3>
              find, save & annotate articles
            </h3>
            <nav>
              <Link
                replace={false}
                to="/"
              >
                Article Search
              </Link>
              <Link
                replace={false}
                to="/saved"
              >
                Saved Articles
              </Link>
            </nav>
          </header>,
          <Switch>
            <Route
              component={[Function]}
              path="/saved"
            />
            <Route
              component={[Function]}
              path="/"
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                New York Times Search
              </h1>,
              <h3>
                find, save & annotate articles
              </h3>,
              <nav>
                <Link
                  replace={false}
                  to="/"
                >
                  Article Search
                </Link>
                <Link
                  replace={false}
                  to="/saved"
                >
                  Saved Articles
                </Link>
              </nav>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New York Times Search",
              },
              "ref": null,
              "rendered": "New York Times Search",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "find, save & annotate articles",
              },
              "ref": null,
              "rendered": "find, save & annotate articles",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/"
                  >
                    Article Search
                  </Link>,
                  <Link
                    replace={false}
                    to="/saved"
                  >
                    Saved Articles
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Article Search",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Article Search",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Saved Articles",
                    "replace": false,
                    "to": "/saved",
                  },
                  "ref": null,
                  "rendered": "Saved Articles",
                  "type": [Function],
                },
              ],
              "type": "nav",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/saved"
              />,
              <Route
                component={[Function]}
                path="/"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/saved",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <header>
            <h1>
              New York Times Search
            </h1>
            <h3>
              find, save & annotate articles
            </h3>
            <nav>
              <Link
                replace={false}
                to="/"
              >
                Article Search
              </Link>
              <Link
                replace={false}
                to="/saved"
              >
                Saved Articles
              </Link>
            </nav>
          </header>
          <Switch>
            <Route
              component={[Function]}
              path="/saved"
            />
            <Route
              component={[Function]}
              path="/"
            />
          </Switch>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header>
              <h1>
                New York Times Search
              </h1>
              <h3>
                find, save & annotate articles
              </h3>
              <nav>
                <Link
                  replace={false}
                  to="/"
                >
                  Article Search
                </Link>
                <Link
                  replace={false}
                  to="/saved"
                >
                  Saved Articles
                </Link>
              </nav>
            </header>,
            <Switch>
              <Route
                component={[Function]}
                path="/saved"
              />
              <Route
                component={[Function]}
                path="/"
              />
            </Switch>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  New York Times Search
                </h1>,
                <h3>
                  find, save & annotate articles
                </h3>,
                <nav>
                  <Link
                    replace={false}
                    to="/"
                  >
                    Article Search
                  </Link>
                  <Link
                    replace={false}
                    to="/saved"
                  >
                    Saved Articles
                  </Link>
                </nav>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New York Times Search",
                },
                "ref": null,
                "rendered": "New York Times Search",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "find, save & annotate articles",
                },
                "ref": null,
                "rendered": "find, save & annotate articles",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Link
                      replace={false}
                      to="/"
                    >
                      Article Search
                    </Link>,
                    <Link
                      replace={false}
                      to="/saved"
                    >
                      Saved Articles
                    </Link>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Article Search",
                      "replace": false,
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": "Article Search",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Saved Articles",
                      "replace": false,
                      "to": "/saved",
                    },
                    "ref": null,
                    "rendered": "Saved Articles",
                    "type": [Function],
                  },
                ],
                "type": "nav",
              },
            ],
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  path="/saved"
                />,
                <Route
                  component={[Function]}
                  path="/"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/saved",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`should render without exploding 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <header>
          <h1>
            New York Times Search
          </h1>
          <h3>
            find, save & annotate articles
          </h3>
          <nav>
            <Link
              replace={false}
              to="/"
            >
              Article Search
            </Link>
            <Link
              replace={false}
              to="/saved"
            >
              Saved Articles
            </Link>
          </nav>
        </header>
        <Switch>
          <Route
            component={[Function]}
            path="/saved"
          />
          <Route
            component={[Function]}
            path="/"
          />
        </Switch>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1>
              New York Times Search
            </h1>
            <h3>
              find, save & annotate articles
            </h3>
            <nav>
              <Link
                replace={false}
                to="/"
              >
                Article Search
              </Link>
              <Link
                replace={false}
                to="/saved"
              >
                Saved Articles
              </Link>
            </nav>
          </header>,
          <Switch>
            <Route
              component={[Function]}
              path="/saved"
            />
            <Route
              component={[Function]}
              path="/"
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                New York Times Search
              </h1>,
              <h3>
                find, save & annotate articles
              </h3>,
              <nav>
                <Link
                  replace={false}
                  to="/"
                >
                  Article Search
                </Link>
                <Link
                  replace={false}
                  to="/saved"
                >
                  Saved Articles
                </Link>
              </nav>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New York Times Search",
              },
              "ref": null,
              "rendered": "New York Times Search",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "find, save & annotate articles",
              },
              "ref": null,
              "rendered": "find, save & annotate articles",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/"
                  >
                    Article Search
                  </Link>,
                  <Link
                    replace={false}
                    to="/saved"
                  >
                    Saved Articles
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Article Search",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Article Search",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Saved Articles",
                    "replace": false,
                    "to": "/saved",
                  },
                  "ref": null,
                  "rendered": "Saved Articles",
                  "type": [Function],
                },
              ],
              "type": "nav",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/saved"
              />,
              <Route
                component={[Function]}
                path="/"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/saved",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <header>
            <h1>
              New York Times Search
            </h1>
            <h3>
              find, save & annotate articles
            </h3>
            <nav>
              <Link
                replace={false}
                to="/"
              >
                Article Search
              </Link>
              <Link
                replace={false}
                to="/saved"
              >
                Saved Articles
              </Link>
            </nav>
          </header>
          <Switch>
            <Route
              component={[Function]}
              path="/saved"
            />
            <Route
              component={[Function]}
              path="/"
            />
          </Switch>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header>
              <h1>
                New York Times Search
              </h1>
              <h3>
                find, save & annotate articles
              </h3>
              <nav>
                <Link
                  replace={false}
                  to="/"
                >
                  Article Search
                </Link>
                <Link
                  replace={false}
                  to="/saved"
                >
                  Saved Articles
                </Link>
              </nav>
            </header>,
            <Switch>
              <Route
                component={[Function]}
                path="/saved"
              />
              <Route
                component={[Function]}
                path="/"
              />
            </Switch>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  New York Times Search
                </h1>,
                <h3>
                  find, save & annotate articles
                </h3>,
                <nav>
                  <Link
                    replace={false}
                    to="/"
                  >
                    Article Search
                  </Link>
                  <Link
                    replace={false}
                    to="/saved"
                  >
                    Saved Articles
                  </Link>
                </nav>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New York Times Search",
                },
                "ref": null,
                "rendered": "New York Times Search",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "find, save & annotate articles",
                },
                "ref": null,
                "rendered": "find, save & annotate articles",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Link
                      replace={false}
                      to="/"
                    >
                      Article Search
                    </Link>,
                    <Link
                      replace={false}
                      to="/saved"
                    >
                      Saved Articles
                    </Link>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Article Search",
                      "replace": false,
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": "Article Search",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Saved Articles",
                      "replace": false,
                      "to": "/saved",
                    },
                    "ref": null,
                    "rendered": "Saved Articles",
                    "type": [Function],
                  },
                ],
                "type": "nav",
              },
            ],
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  path="/saved"
                />,
                <Route
                  component={[Function]}
                  path="/"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/saved",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
